src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -471,6 +473,20 @@ protected override void Disposing()
                     if (journal.Value.IsValueCreated)
                         journal.Value.Value.Dispose();
                 }
+
+                lock (_journalsForReuse)
+                {
+                    foreach (var reusableFile in _journalsForReuse.Values)
```

Comment:
This make me wonder if we wants this in this manner. Imagine a case where you are shutting down a busy server, you don't want to lose these files. 

Wouldn't it be better to use it with a flag that will prevent pending recycle  in `RecoverDatabase`?